### ISSUE TITLE
Buying Brax Bug.

### ISSUE THREAD

**Issue by Questiondeca** [Innoxia/liliths-throne-public#1195]
A Flag or trigger is not being set/fired properly during the quest to buy Brax (Bree).

Upon giving Candi her Perfume from Succubus Secrets, she sends you out to get illegal Harpy Lollipops from the Harpy Nests Enforcer Outpost, however, upon arrival no option appears to gain the lollipops.

To Reproduce, attempt to buy Brax, start the quest, and go to the Harpy Nests after turning in the Perfume.

 - Game version [e.g. 0.3.4]
 - OS: [ Windows 10]
 - Using .jar
- Java version: [1.8.0_221]

**Comment by Ian-Luminus** [comment]
And if you start a new game and do this quest before you open up the Harpy Nests, the Harpy Lollipops are available.. but you lose the dialogue option to proceed into the nests after talking to Scarlett, unless there's some other prerequisite I'm forgetting.

tl;dr my new save has Bree as mine but cannot access the nests now.

**Comment by Questiondeca** [comment]
Manually spawning in the lollipops with the debug tool also allows you to complete that stage of the quest. So the problem is probably in HarpyNestsDialogue.java , specifically the code that sets what options are available. It properly recognizes that whether or not you've completed the Harpy Nests Access sidequest (and whether you've pacified the Queens), but obviously it's mixing up the two sidequests... but they are adjacent in HarpyNestsDialogue.java so perhaps a bracketing error?

**Comment by Questiondeca** [comment]
A, less important bug, if you put Bree in your party, combat text ignores any gender and species changes and refers only to Brax and Wolf.

**Comment by Questiondeca** [comment]
One final bug before going to bed, Resetting Bree's Perks removes only some of their perks, but also sets their perk points to 0, not refunding any points gained by level.

**Comment by Ian-Luminus** [comment]
I'm not sure if changing from Index to Array while Index 1 was already set is what did it.. but line 166 in 0.3.3.8 and line 169 in 0.3.4 is where that issue is. I wish I knew Java better to actually contribute to the fix, but hopefully not having to search as much helps some.
